ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/VulnReports/OnDemandReportsTab.tsx

```diff
@@ -34,10 +134,53 @@ function OnDemandReportsTab() {
             <PageSection>
                 <Card>
                     <CardBody className="pf-v5-u-p-0">
+                        <Toolbar>
+                            <ToolbarContent>
+                                <ToolbarItem alignItems="center">
+                                    <ReportJobStatusFilter
+                                        availableStatuses={[
+                                            'WAITING',
+                                            'PREPARING',
+                                            'DOWNLOAD_GENERATED',
+                                            'ERROR',
+                                        ]}
+                                        selectedStatuses={ensureReportJobStatuses(
+                                            reportJobStatusFilters
+                                        )}
+                                        onChange={onReportJobStatusFilterChange}
+                                    />
+                                </ToolbarItem>
+                                <ToolbarItem className="pf-v5-u-flex-grow-1" alignSelf="center">
+                                    <MyJobsFilter
+                                        isViewingOnlyMyJobs={ensureBoolean(isViewingOnlyMyJobs)}
+                                        onMyJobsFilterChange={onMyJobsFilterChange}
+                                    />
+                                </ToolbarItem>
+                                <ToolbarItem variant="pagination" align={{ default: 'alignRight' }}>
+                                    <Pagination
+                                        toggleTemplate={({ firstIndex, lastIndex }) => (
+                                            <span>
+                                                <b>
+                                                    {firstIndex} - {lastIndex}
+                                                </b>{' '}
+                                                of <b>many</b>
+                                            </span>
```

Comment:
I know there has been discussion on reducing the number of `count` API calls for performance reasons, but I have heard a handful of complaints on the confusing UX of the indeterminate pagination pattern. It might be worthwhile comparing the performance impact of getting a strict count versus the UX loss of this approach.